### DISCUSSION TITLE
How can I find out the `python_functions` value before running tests ?

### DISCUSSION THREAD

**Discussion by nicos68** [pytest-dev/pytest#11950] [Q&A] [answered]
I reported [this issue](https://github.com/nvim-neotest/neotest-python/issues/65) on the neotest-python neovim plugin. I could code the improvement myself but I don’t see any way to get the `python_functions` configuration value from pytest. Is there any way I could get it without writing a test (which I can’t since I don’t know this value) ?

**Comment by nicoddemus** [comment]
> Is there any way I could get it without writing a test (which I can’t since I don’t know this value) ?

Can you give more information? For example, can you use Python code which is executed in a Python environment with pytest installed? Or you want to find that information just by looking at files?

**Reply by nicos68** [reply]
My intention is precisely to use pytest to find that out to avoid bugs or discrepancies between the plugin and pytest's implementation, so yes I can execute python code in a python environment (as you can see is already done [here](https://github.com/nvim-neotest/neotest-python/blob/master/neotest_python/pytest.py)).

**Reply by RonnyPfannschmidt** [reply]
A plugin can just fetch the option value

**Reply by nicos68** [reply]
So just to make sure I understand this correctly, you're suggesting my best bet is to reimplement reading the value from `pyproject.toml`/`pytest.ini`/`setup.cfg`/... ?

I wanted to avoid doing that because changes in pytest's implementation would cause impacts in the plugin but if that's the only solution then sure I can do it.

**Reply by RonnyPfannschmidt** [reply, marked as answer]
Uhm No, just use the pytest Config object post parsing in a pytest plugin

**Reply by nicos68** [reply]
Thanks I sent them a [PR](https://github.com/nvim-neotest/neotest-python/pull/66).
